File: 김은지/방문-길이.js
const ROW_RANGE = { MIN: -5, MAX: 5 };
const COL_RANGE = { MIN: -5, MAX: 5 };

const NEXT_DIRECTION = {
  U: [-1, 0],
  D: [1, 0],
  R: [0, 1],
  L: [0, -1],
};

function outOfRange(row, col) {
  return row < ROW_RANGE.MIN || row > ROW_RANGE.MAX || col < COL_RANGE.MIN || col > COL_RANGE.MAX;
}

function solution(dirs) {
  const path = new Set();

  let currPoint = [0, 0];
  let cnt = 0;
  [...dirs].forEach((dir) => {
    const [currRow, currCol] = currPoint;
    const [dRow, dCol] = NEXT_DIRECTION[dir];
    const [nextRow, nextCol] = [currRow + dRow, currCol + dCol];

    if (outOfRange(nextRow, nextCol)) return;

    currPoint = [nextRow, nextCol];

    let key;
    if (currRow < nextRow || currCol < nextCol) {
      key = `[${currRow}, ${currCol}] - [${nextRow}, ${nextCol}]`;
    } else {
      key = `[${nextRow}, ${nextCol}] - [${currRow}, ${currCol}]`;
    }
    if (path.has(key)) return;
    path.add(key);
    cnt++;
  });

  return cnt;
}

File: 김현경/방문_길이.py
def solution(dirs):
    visited = set()
    dicts = {'U': (0, 1), 'D': (0, -1), 'L': (-1, 0), 'R': (1, 0)}
    x, y = 0, 0 # 초기 좌표 (0, 0) 
    
    for i in dirs:
        ix, iy = dicts[i][0] + x, dicts[i][1] + y
        
        # 범위를 벗어나지 않는 경우
        if -5 <= ix <= 5 and -5 <= iy <= 5:
            visited.add(((x, y), (ix, iy)))
            visited.add(((ix, iy), (x, y)))
            x, y = ix, iy

    return len(visited) // 2 # 양방향 경로 모두 저장했기 때문에 2로 나눠줌